rfd/0236-access-list-preset.md

```diff
@@ -0,0 +1,263 @@
+
+#### Dynamic Feedback
+
+The guide will provide dynamic feedback as the admin is defining/tweaking access to resources in the form of:
+
+- Listing the resources given some admin input. E.g. if an admin was defining access for applications, and the admin specifies label `env: staging`, the feedback is to list only applications with matching label.
+- If the cluster has `Identity Security` feature enabled, the admin will also be able to use access graph to see a graph of accessible resources.
+
+The dynamic feedback allows admin to essentially double check and preview what members of this access list will see in their own account.
+
+#### Terraform Support
+
+The guide will provide a copyable Terraform script that is equivalent to going through the guide to create an access list with a preset. At the end of the guide, the admin can choose to use the Terraform script only, or let the web app proceed to finish the request.
+
+Similar to dynamic feedback, as the admin is defining/tweaking inputs in the guide, the Terraform script will dynamically update to reflect the latest changes. This script will be visible throughout the entire guide (can optionally be hidden).
+
+The Terraform script will also be available when viewing an access list created with a preset.
+
+### Security
+
+To prevent users (admins) from overstepping the boundaries set by their assigned roles, the following restrictions are placed when user wants to perform create, delete, or update operations on an access list created with a preset:
+
+- The user is required to have access to perform all CRUD operations on a role. The user may not be directly performing actions on a role, but Teleport will on behalf of the user.
+- The [same existing rules](https://github.com/gravitational/teleport.e/blob/e49a5ad654408ce0779622c38c7acda0417bfef0/lib/accesslist/service.go#L1648) are applied when creating/deleting/updating an access list. E.g. for updating an access list the user will have to either be a owner or have access to `access_list` resource rules.
+- When defining or modifying access to resources for an access list, the user is limited to what their own roles allow them. For example, lets say a few application resources exists in a Teleport cluster:
+  - If the users currently assigned role does not define access to applications (missing `app_labels`) and therefore unable to list applications, the user will not be able to define any access to application.
+  - If the users currently assigned role defines a `app_label` giving access to just `some` of the applications, the user will be only able to define access to the resources they can see.
+  - If an access list with preset was created, and later a user with lesser permission than the original creator comes in to edit the access to resources, the access definition for resources that produced no results will be removed on save. This prevents defining random labels.
```

Comment:
this is a TODO, this is the proposed workflow until we can agree on a way for a admin to list all resources